settings/settings.go

```diff
@@ -227,3 +276,312 @@ func objectValidator(bound int, args ...string) bool {
+			}
+		}
+		if parseErr {
+			return errors.New("there was an error during export")
+		}
+		return nil
+	}
+
+	return errors.New("Command must be used with Airflow 2.X")
+}
+
+func ExportConnections(id string, logs bool) error {
+	// Setup airflow command to export connections
+	airflowCommand := "airflow connections list -o yaml"
+	out := execAirflowCommand(id, airflowCommand)
+	if logs {
+		fmt.Println("Export Connections logs:\n" + out)
+	}
+	// remove all color from output of the airflow command
+	plainOut := re.ReplaceAllString(out, "")
+	// remove extra warning text
+	yamlCons := "- conn_id:" + strings.SplitN(plainOut, "- conn_id:", 2)[1]
+
+	var connections ListConnections
+
+	err := yaml.Unmarshal([]byte(yamlCons), &connections)
+	if err != nil {
+		return err
+	}
+	// add connections to settings file
+	for i := range connections {
+		port, err := strconv.Atoi(connections[i].ConnPort)
+		if err != nil {
+			fmt.Println("Issue with parsing port number: ")
+			fmt.Println(err)
+		}
+		fmt.Println("Exporting Connection: " + connections[i].ConnID)
+
+		newConnection := Connection{
+			Conn_ID:       connections[i].ConnID,
+			Conn_Type:     connections[i].ConnType,
+			Conn_Host:     connections[i].ConnHost,
+			Conn_Schema:   connections[i].ConnSchema,
+			Conn_Login:    connections[i].ConnLogin,
+			Conn_Password: connections[i].ConnPassword,
+			Conn_Port:     port,
+			Conn_URI:      connections[i].ConnURI,
+			Conn_Extra:    connections[i].ConnExtra,
+		}
+
+		settings.Airflow.Connections = append(settings.Airflow.Connections, newConnection)
+	}
+	// write to settings file
+	viperSettings.Set("airflow", settings.Airflow)
+	err = viperSettings.WriteConfig()
+	if err != nil {
+		return err
+	}
+	fmt.Printf("successfully exported Connections\n\n")
+	return nil
+}
+
+func ExportVariables(id string, logs bool) error {
+	// setup files
+	airflowCommand := "airflow variables export tmp.var"
+	out := execAirflowCommand(id, airflowCommand)
+	if logs {
+		fmt.Println("Export Variables logs:\n" + out)
+	}
+
+	if strings.Contains(out, "successfully") {
+		// get variables created by the airflow command
+		fileCmd := "cat tmp.var"
+		out = execAirflowCommand(id, fileCmd)
+
+		m := map[string]string{}
+		err := json.Unmarshal([]byte(out), &m)
+		if err != nil {
+			fmt.Println("variable json decode unsuccessful")
+		}
+		// add the variables to settings object
+		for k, v := range m {
+			newVariables := Variables{{k, v}}
+			fmt.Println("Exporting Variable: " + k)
+			settings.Airflow.Variables = append(settings.Airflow.Variables, newVariables...)
+		}
+		// write variables to settings file
+		viperSettings.Set("airflow", settings.Airflow)
+		err = viperSettings.WriteConfig()
+		if err != nil {
+			return err
+		}
+		rmCmd := "rm tmp.var"
+		_ = execAirflowCommand(id, rmCmd)
+		fmt.Printf("successfully exported variables\n\n")
+		return nil
+	}
+	return errors.New("variable export unsuccessful")
+}
+
+func ExportPools(id string, logs bool) error {
```

Comment:
can we combine the common logic of this method & EnvExportVariables, because most of it is the exact same, except where the output is stored

Reply:
I actually think this function is rally different from EnvExportVariables. the parsing of the output is completely different which is most of the function